server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -184,7 +189,12 @@ public static class UploadedSegmentMetadata {
 
         @Override
         public String toString() {
-            return String.join(SEPARATOR, originalFilename, uploadedFilename, checksum, String.valueOf(length));
+            String metadataStr = String.join(SEPARATOR, originalFilename, uploadedFilename, checksum, String.valueOf(length));
+            if (writtenBy != null) {
```

Comment:
In what scenarios `writtenBy` will be null?

Reply:
This would be the cases where `toString( )` gets invoked directly or indirectly before the `writtenBy` is set by the setter. This is an extra check to avoid transient failures.

Reply:
nit - can we do `String metadataStr = String.join(SEPARATOR, originalFilename, uploadedFilename, checksum, String.valueOf(length), String.valueOf(writtenBy));`. This way we can skip the condition. When the writtenBy value is null, it would show up as `blab-blah..-null` and when not null, the valid value would show up.

Reply:
@ashking94 what is the value addition by adding a `null` at the end? In which case could it be useful?

Reply:
IMO it's okay to return the null value in the toString (stays coherent with other cases where it is non null), avoid the if condition and have minimum lines of code.